core/block/detailservice/set_details.go

```diff
@@ -0,0 +1,248 @@
+			Key:   k,
+			Value: v,
+		})
+	}
+	return s.SetDetails(nil, workspaceId, setDetails)
+}
+
+func (s *service) SetSource(ctx session.Context, objectId string, source []string) error {
+	return cache.Do(s, objectId, func(sb smartblock.SmartBlock) error {
+		st := sb.NewStateCtx(ctx)
+		// nolint:errcheck
+		_ = st.Iterate(func(b simple.Block) (isContinue bool) {
+			if dv := b.Model().GetDataview(); dv != nil {
+				for _, view := range dv.Views {
+					view.DefaultTemplateId = ""
+					view.DefaultObjectTypeId = ""
+				}
+				st.Set(b)
+				return false
+			}
+			return true
+		})
+		st.SetDetailAndBundledRelation(bundle.RelationKeySetOf, pbtypes.StringList(source))
+
+		flags := internalflag.NewFromState(st)
+		// set with source is no longer empty
+		flags.Remove(model.InternalFlag_editorDeleteEmpty)
+		flags.AddToState(st)
+
+		return sb.Apply(st, smartblock.NoRestrictions, smartblock.KeepInternalFlags)
+	})
+}
+
+func (s *service) SetWorkspaceDashboardId(ctx session.Context, workspaceId string, id string) (setId string, err error) {
+	err = cache.Do(s, workspaceId, func(ws *editor.Workspaces) error {
+		if ws.Type() != coresb.SmartBlockTypeWorkspace {
+			return ErrUnexpectedBlockType
+		}
+		if err = ws.SetDetails(ctx, []*model.Detail{
+			{
+				Key:   bundle.RelationKeySpaceDashboardId.String(),
+				Value: pbtypes.String(id),
+			},
+		}, false); err != nil {
+			return err
+		}
+		return nil
+	})
+	return id, err
+}
+
+func (s *service) SetIsFavorite(objectId string, isFavorite bool) error {
+	spaceID, err := s.resolver.ResolveSpaceID(objectId)
+	if err != nil {
+		return fmt.Errorf("resolve spaceID: %w", err)
+	}
+	spc, err := s.spaceService.Get(context.Background(), spaceID)
+	if err != nil {
+		return fmt.Errorf("get space: %w", err)
+	}
+	return s.objectLinksCollectionModify(spc.DerivedIDs().Home, objectId, isFavorite)
+}
+
+func (s *service) SetIsArchived(objectId string, isArchived bool) error {
+	spaceID, err := s.resolver.ResolveSpaceID(objectId)
+	if err != nil {
+		return fmt.Errorf("resolve spaceID: %w", err)
+	}
+	spc, err := s.spaceService.Get(context.Background(), spaceID)
+	if err != nil {
+		return fmt.Errorf("get space: %w", err)
+	}
+	if err := s.checkArchivedRestriction(isArchived, objectId); err != nil {
+		return err
+	}
+	return s.objectLinksCollectionModify(spc.DerivedIDs().Archive, objectId, isArchived)
+}
+
+func (s *service) SetListIsFavorite(objectIds []string, isFavorite bool) error {
+	ids, err := s.store.HasIDs(objectIds...)
+	if err != nil {
+		return err
+	}
+	var (
+		anySucceed  bool
+		resultError error
+	)
+	for _, id := range ids {
+		err := s.SetIsFavorite(id, isFavorite)
+		if err != nil {
+			log.Error("failed to favorite object", zap.String("objectId", id), zap.Error(err))
+			resultError = errors.Join(resultError, err)
+		} else {
+			anySucceed = true
+		}
+	}
+	if resultError != nil {
+		log.Warn("failed to set objects as favorite", zap.Error(resultError))
+	}
+	if anySucceed {
```

Comment:
The same comment as for isArchive function

Reply:
Same situation is here